auctions/dutch-auction/src/abi.sw

```diff
@@ -0,0 +1,16 @@
+library abi;
+
+dep data_structures;
+
+use std::{contract_id::ContractId, identity::Identity};
+
+use data_structures::Auction;
+
+abi DutchAuction {
```

Comment:
can you add a function that allows only the beneficiary of the auction to edit the auction if no one has bid and the auction has not ended? This would allow them to extend the end time, change the asset to be sent and the beneficiary to be someone else. Perform any necessary error handling and any other changes that you think are valid

Reply:
I think it will also be useful to expose the `auction_count` similar to the `price`. No error handling required

Reply:
All extending the end time would do is increase the price and slow down the price decline. Users can still bid on an auction past the end_time aslong as the auction isnt cancelled (ended). Only useful thing to change is the beneficiary and I suppose the asset_id if maybe the wrong one was set or something like that

Reply:
Shouldn't the auction automatically end when the end time is passed and someone tries to do anything?

Reply:
thats a design choice which could be argued for either way. My thought process was that since the price is going down to a reserve price (the minimum the seller will sell for), the auctioneer would be fine with getting that sale even after the end_time, or else the reserve price would have been higher

Reply:
I just made it so people cant bid after end_time, although some people might want the sale, its still unexpected behaviour i suppose.